mcode/mcode-extraction-framework
# CancerRelatedMedication STU2 Updates

# Summary
CancerRelatedMedicationExtractor has been converted to CancerRelatedMedicationAdministrationExtractor and a new CancerRelatedMedicationRequestExtractor has been added. 

Profiles:
https://build.fhir.org/ig/HL7/fhir-mCODE-ig/branches/master/StructureDefinition-mcode-cancer-related-medication-request.html
http://hl7.org/fhir/us/mcode/2021May/StructureDefinition-mcode-cancer-related-medication-administration.html

## New behavior
CancerRelatedMedicationAdministrationExtractor behaves the same as the old CancerRelatedMedicationExtractor. It's pretty much just a renaming. CancerRelatedMedicationRequestExtractor is a new extractor based on the profile from STU2 linked above.
## Code changes
- All files associated with the previous `CancerRelatedMedicationExtractor` (the extractor, template and tests) have been renamed to  `CancerRelatedMedicationAdministration` to distinguish them from `CancerRelatedMedicationRequest`
- New extractor, template and tests have been added for `CancerRelatedMedicationRequest` profile
- `MCODEClient.js`, `index.js` and `ResourceGenerator.js` updated to support the new and renamed extractors
- Updated sample client data to have new `cancer-related-medication-request-information.csv` and renamed `cancer-related-medication-administration-information.csv`

# Testing guidance
- Ensure that in converting CancerRelatedMedicationStatement to CancerRelatedMedicationAdministration, I didn't miss any other changes made in STU2
- Check that my implementation of CancerRelatedMedicationRequest properly implements the profile (if I am missing any fields or implementing any fields that aren't necessary) and check that the new CSV format makes sense
- Ensure that the client runs and tests pass
- Here is an example config for the relevant extractors:
```
"extractors": [
    {
      "label": "patient",
      "type": "CSVPatientExtractor",
      "constructorArgs": {
        "filePath": "./test/sample-client-data/patient-information.csv"
      }
    },
    {
      "label": "cancerRelatedMedicationAdministration",
      "type": "CSVCancerRelatedMedicationAdministrationExtractor",
      "constructorArgs": {
        "filePath": "./test/sample-client-data/cancer-related-medication-administration-information.csv"
      }
    },
    {
      "label": "cancerRelatedMedicationRequest",
      "type": "CSVCancerRelatedMedicationRequestExtractor",
      "constructorArgs": {
        "filePath": "./test/sample-client-data/cancer-related-medication-request-information.csv"
      }
    }
  ]
```
